test/unittests/capi_test.cpp

```diff
@@ -98,6 +98,45 @@ TEST(capi, find_exported_function_index)
     fizzy_free_module(module);
 }
 
+TEST(capi, find_exported_function)
+{
+    /* wat2wasm
+    (module
+      (func $f (export "foo") (result i32) (i32.const 42))
+      (global (export "g1") i32 (i32.const 42))
+      (table (export "tab") 10 30 anyfunc)
+      (memory (export "mem") 1 2)
+    )
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000105016000017f0302010004050170010a1e0504010101020606017f00412a0b0718040366"
+        "6f6f00000267310300037461620100036d656d02000a06010400412a0b");
+
+    auto module = fizzy_parse(wasm.data(), wasm.size());
+    ASSERT_NE(module, nullptr);
+
+    auto instance = fizzy_instantiate(module, nullptr, 0, nullptr, nullptr, nullptr, 0);
+    ASSERT_NE(instance, nullptr);
+
+    FizzyExternalFunction function;
+    FizzyExportedFunctionContext* context;
+    ASSERT_TRUE(fizzy_find_exported_function(instance, "foo", &function, &context));
+    EXPECT_EQ(function.type.inputs_size, 0);
+    EXPECT_EQ(function.type.output, FizzyValueTypeI32);
+    EXPECT_EQ(function.context, context);
```

Comment:
Hmm, why do we need to return `FizzyExportedFunctionContext` if the context is available here as well?

Reply:
Yeah, we return it twice - as `void*` opaque external function context and as `FizzyExportedFunctionContext` for convenience.

Also not completely sure it's a good idea, but without this we either require the user to do the cast to `FizzyExportedFunctionContext*` themselves, or make "free" function less type-safe like `fizzy_free_exported_function_context(void*)`.

Reply:
But why not make the function expect a `void*` so that no cast is needed, and introduce `fizzy_free_exported_function(FizzyExternalFunction)`?

This was the `FizzyExternalFunction` is fully symmetric whether it is defined as a host function or acquired as an import. Can be called the same way, can be passed the same way.

The only safety thing the user has to observer is that `fizzy_free_exported_function()` is UB if they call it on something not acquired via `fizzy_find_exported_function()`.

What do you think @gumb0 @chfast ?

Reply:
> Also not completely sure it's a good idea, but without this we either require the user to do the cast to `FizzyExportedFunctionContext*` themselves, or make "free" function less type-safe like `fizzy_free_exported_function_context(void*)`.

Actually in this variant cast would be implicit, so it's either
`fizzy_free_exported_function_context(FizzyExportedFunctionContext*)` and user would pass `void*` there
or `fizzy_free_exported_function_context(void*)`

Reply:
I like the idea with `fizzy_free_exported_function(FizzyExternalFunction*)`, it's simpler.

Only need to clarify in documentation that it should be used with exported functions only.